### PR TITLE
RAC4861: SprintTag missed logs of the first 9 times test

### PR DESCRIPTION
**Pull request by PengTian0**
PR for [RAC4861](https://rackhd.atlassian.net/browse/RAC-4861)
To fix the issue, the PR moves logs of unit test to ```UnitTest${i}``` and logs of function test to ```FunctionTest${i}``` in SprintTag and SprintRelease.

The artifacts looks like:
image

### CODE REVIEW

**Comment by PengTian0** [comment]
@panpan0000 @changev @anhou @yyscamper

**Comment by PengTian0** [comment]
test this please

File: jobs/FunctionTest/FunctionTest.groovy [hunk 1]
```diff
@@ -0,0 +1,210 @@
+import groovy.transform.Field;
```
Comment (panpan0000) [inline jobs/FunctionTest/FunctionTest.groovy:1]: Per @PengTian0 , this is almost just change the file name and refine the code layout.
so assuming it functions the same as original code

File: jobs/ContinuousTest/Jenkinsfile [hunk 2]
```diff
@@ -52,7 +52,14 @@ node{
         }
```
Comment (panpan0000) [inline jobs/ContinuousTest/Jenkinsfile:52]: Can you please summarize the change key points of your PR ?
it's not small and not easy to review without some guidance

File: jobs/SprintTag/Jenkinsfile [hunk 3]
```diff
@@ -16,27 +16,37 @@ node{
...
             }
             stage("Function Test${i}"){
-                load("jobs/function_test/function_test.groovy")
+                function_test.run(TESTS, env.stash_manifest_name, env.stash_manifest_path, repo_dir)
             }
         } catch(error){
             echo "Caught: ${error}"
             failed_times += 1
+        } finally{
+            unit_test.archiveArtifactsToTarget("UnitTest${i}")
```
Comment (panpan0000) [inline jobs/SprintTag/Jenkinsfile:40]: Can you paste a snapshot of what's the final view of the SprintTag build artifacts ?

File: jobs/UnitTest/UnitTest.groovy [hunk 4]
```diff
@@ -28,59 +28,80 @@ def setTestRepos(test_repos){
...
         }
     }
 }
 
+def archiveArtifactsToTarget(target){
+    try{
+        dir("$target"){
+            for(int i=0; i<test_repos.size; i++){
+                def repo_name = test_repos.get(i)
+                unstash "$repo_name"
```
Comment (panpan0000) [inline jobs/UnitTest/UnitTest.groovy]: per @PengTian0 , the argument of "$repo_name" should align the  name value in line 45 ```stash name: "$repo_name", includes: 'xunit-reports/*.xml'``` . 
it's a coupling between two functions.  just curious, is it possible to create  a class member variable to align the value of these 2 position ?

File: jobs/pr_gate/pr_parser.groovy [hunk 5]
```diff
@@ -24,6 +24,7 @@ node{
             archiveArtifacts 'manifest'
             stash name: 'manifest', includes: 'manifest'
             env.stash_manifest_name = "manifest"
+            env.stash_manifest_path = "manifest"
```
Comment (panpan0000) [inline jobs/pr_gate/pr_parser.groovy:27]: ?? what's the difference between name & path.
why they are in the same value ?

File: jobs/SprintTag/Jenkinsfile (hunk 3 above)
Comment (PengTian0) [reply jobs/SprintTag/Jenkinsfile:40]: Sure~

**Comment by PengTian0** [comment]
test this please

**Comment by panpan0000** [comment]
Thank you to add more comments. @PengTian0 
Now :+1 for me

**Review by panpan0000 — APPROVED** [review]

File: jobs/pr_gate/pr_parser.groovy (hunk 5 above)
Comment (PengTian0) [reply jobs/pr_gate/pr_parser.groovy:27]: The ```name``` is a ```name``` of ```stash```
The ```path``` is the saved files by the ```stash``` step

File: jobs/ContinuousTest/Jenkinsfile (hunk 2 above)
Comment (PengTian0) [reply jobs/ContinuousTest/Jenkinsfile:52]: Sure, I already added more comments to the PR

**Comment by PengTian0** [comment]
test this please

**Review by anhou — APPROVED** [review]

**Pull request merged**